java/datastructures/lib/src/main/java/codechallenges/linkedlist/AnimalShelter.java

```diff
@@ -0,0 +1,76 @@
+package codechallenges.linkedlist;
+
+
+
+
+import datastructures.queue.Queue;
+
+public class AnimalShelter {
+
+  // Animal class
+  public static class Animal {
+    private String name;
+    private AnimalType type;
+
+    public Animal(String name, AnimalType type) {
+      this.name = name;
+      this.type = type;
+    }
+
+    public String getName() {
+      return name;
+    }
+
+    public AnimalType getType() {
+      return type;
+    }
+  }
+
+  // Animal type enum
+  public enum AnimalType {
+    DOG,
+    CAT
+  }
+
+  // Fields
+  private Queue<Animal> dogQueue;
+  private Queue<Animal> catQueue;
+
+  // Constructor
+  public AnimalShelter() {
+    dogQueue = new Queue<>();
+    catQueue = new Queue<>();
+  }
+
+  // Enqueue method
+  public void enqueue(Animal animal) {
+    if (animal.getType() == AnimalType.DOG) {
+      dogQueue.enqueue(animal);
+    } else if (animal.getType() == AnimalType.CAT) {
+      catQueue.enqueue(animal);
+    }
+  }
+
+  // Dequeue method
+  public Animal dequeue(AnimalType pref) {
+    if (pref == AnimalType.DOG && !dogQueue.isEmpty()) {
+      return dogQueue.dequeue();
+    } else if (pref == AnimalType.CAT && !catQueue.isEmpty()) {
+      return catQueue.dequeue();
+    } else if (pref == null) {
+      // Return the oldest animal if no preference is specified
+      if (!dogQueue.isEmpty() && !catQueue.isEmpty()) {
+        if (dogQueue.peek().getName().compareTo(catQueue.peek().getName()) < 0) {
```

Comment:
You're on the right track here, but comparing strings is tricky, and you're going to have to use a lot of workarounds to get the data you want. 

You'll need a way to track the order that each animal was added, independent of any of its other properties. One option would be to add an integer field to your Animal class that tracks the order they were created in. Another would be to redo the AnimalShelter class so it only uses one queue, and find a different way to sort between cats and dogs. I'm sure there are other ways to do it - feel free to experiment!